group_set.go

```diff
@@ -0,0 +1,32 @@
+package sheriff
+
+type groupSet map[string]int
+
+func (s groupSet) IncrementGroups(groups []string) {
+	for i := range groups {
+		s[groups[i]]++
+	}
+}
+
+func (s groupSet) DecrementGroups(groups []string) {
+	for i := range groups {
+		s[groups[i]]--
+	}
+}
+
+func (s groupSet) Contains(group string) bool {
+	c, ok := s[group]
+	if !ok {
+		return false
+	}
```

Comment:
Just pedantics, but given the return is `(c > 0)` the `ok` idiom technically isn't needed here (if not found, it'll return the default value for the map).  I think this function could be reduced to `return (s[group] > 0)`

Reply:
I believe it still gets inserted the case where it's not in the map and the default value is returned, which is why I did it this way. But I'm not 100% confident that's true.

Reply:
Oh snap, it doesn't! Code updated.